gpkit/signomial_program.py

```diff
@@ -66,8 +69,16 @@ def __init__(self, cost, constraints, verbosity=2):
                     n_cs.append(-c)
                     n_exps.append(exp)
                     self.negvarkeys.update(exp.keys())
-            posy = Posynomial(p_exps, p_cs) if p_cs != [] else None
-            negy = Posynomial(n_exps, n_cs) if n_cs != [1] else None
+            if isinstance(sig, Posynomial):
+                posy = Posynomial(p_exps, p_cs) if p_cs != [] else None
+                negy = Posynomial(n_exps, n_cs) if n_cs != [1] else None
+                assert posy
+                assert not negy
+            else:
+                posy = Posynomial(p_exps, p_cs) if p_cs != [] else None
+                negy = Posynomial(n_exps, n_cs) if n_cs != [] else None
+                assert posy
+                assert negy
```

Comment:
Everything is going to be a constraint, and this separation should happen in those constraints `__init__`